### ISSUE TITLE
Optional_args not working

### ISSUE THREAD

**Issue by omron93** [napalm-automation-community/napalm-sros#41]
The last [commit](https://github.com/napalm-automation-community/napalm-sros/commit/ffc9992ea6f085e821b34bb5885ba933f83ae3ca) broke few thing
- I agree that's a correct approach of using optional args (passing the optional args in constructor -> that can change behavior (e.g. format) of other functions for that driver object)
- But with current implementation the 'optional_args' can't be used to alter `format` in `get_config` and `json_format` in `compare_config`
- Also I believe now when you do following calls then it fails. Calling get_config make compare_config failing and vice versa.
```
device.get_config()
device.compare_config()
```

The reason is that `self.optional_args` should be assigned to only in constructor and in read only in the other functions.

The current implementation is that in constructor the `None` is always assigned to `self.optional_args` -> then get_config checks `self.optional_args is None`, in this case it is so it assigns default value for 'format' to it -> when calling compare_config the `self.optional_args` is not None but it doesn't have 'json_format' key either, so it fails.

A way how to implement this could be to have in the constructor
```
self.optional_args = optional_args
```
and then e.g. in compare_config having
```
json_format = self.optional_args.get("json_format", False)
```
and to use `json_format` variable in checks if to return diff in json format or not

**Comment by JonLundstrom** [comment]
Thanks for catching this and the inputs. We'll dig into it.

**Comment by omron93** [comment]
@JonLundstrom  @ashah26 I've raised PR (#42 ) to fix this. Please review.

**Comment by JonLundstrom** [comment]
@omron93 Thanks for the PR. We are currently testing a fix for this issue, but will review your PR as well. Stay tuned, it'll take a couple days to complete our testing and review.

**Comment by JonLundstrom** [comment]
This issue was addressed by PR #44
